### ISSUE TITLE
Reorg util files

### ISSUE THREAD

**Issue by paulfalgout** [marionettejs/backbone.marionette#2858]
So we have "classes", we have utils, now mixins, and an unnamed group of things.
Those will be:
- bind-entity-events
- bind-radio-events
- error
- monitor-view-events
- renderer
- template-cache
- trigger-method

I also think possibly `utils/deprecate` belongs in this list.

These things to me are more than a simple util, but they're not classes.
Best name I could come up with is module.  It also seems like each of these things could potentially be pulled out as submodule dependencies and be ok.  I got that idea from this: https://github.com/marionettejs/backbone.marionette/issues/2327

Although I suspect eventually triggerMethod will just be part of the `Marionette.Events` mixin... so there are grey areas for sure.. the two binds could also be mixins..

@marionettejs/marionette-core thoughts?

**Comment by jmeas** [comment]
Sounds like a fine idea to me. One note on the name: the word 'module' carries too much meaning in JavaScript, I think. For instance, each individual file in the Marionette repo is a module now, as we're using ES2015 modules. I'd choose any word other than module.

**Comment by JSteunou** [comment]
tools ? extracts ? pieces ? components ?

**Comment by jasonLaster** [comment]
I think many of these could be in utils bind-entity-events, bind-radio,events, monitor-view-events, triggerMethod. the others could stay in the main folder

**Comment by paulfalgout** [comment]
I'm ok with that too.  My only concern is that the utils dir is getting quite large and, other than `deprecate`, is now comprise of only single function utilities.  This re-org could also possibly wait until we land somewhere on documentation.  How we end up describing things may be even more telling.

**Comment by ianmstew** [comment]
I'm used to a `utils` folder containing a number of utility _types_ and each utilities *type* exports a number of utility methods.  ES6 modules/export syntax makes this really nice.  E.g.,

utils/event-util.js
```
export function MonitorViewEvents() ...
export function triggerMethod() ...
export function triggerMethodOn() ...
```

utils/options-util.js
```
export function getOption() ...
export function getValue() ...
export function mergeOptions() ...
```

**Comment by paulfalgout** [comment]
I like that.  
As far as exporting..  You could be adding this to a Bb.View right?

**Comment by ianmstew** [comment]
Cool, that could help with the size of the utils folder.

Hmm, well we are now monitoring view events for Bb.View from within Region#show and CV#addChild.  Does that remove the need?

**Comment by paulfalgout** [comment]
Oh you're totally right.. missed the CollectionView applying the monitor on the buildChildView.  I think that covers everything.

**Comment by jasonLaster** [comment]
cool. nice idea @ianmstew. I think I'd prefer folders as opposed to files, so e.g. `utils/options/mergeOptions` this makes mergeOptions a little bit more discoverable for me and keeps some files from exploding. Ofcourse, this won't be true everywhere, I still imagine `triggerMethod` and `triggerMethodOn` in the same file, but in most cases I'd personally prefer folders. Thoughts?

**Comment by ianmstew** [comment]
@jasonLaster So, it looks like a question of:

```
import { getOption, getValue } from 'utils/options-util'
```
vs.
```
import getOption from 'utils/options/get-option';
import getValue from 'utils/options/get-value';
```

(That may not be the perfect example, but it gets the idea across.)

The logical grouping is the same, it's just a question of whether to use the logical grouping as a utility library name or a folder name.  If one file is blowing up, that means the corresponding folder would be blowing up, too, so the same thought will need to go into the groupings either way.

I have a feeling the answer is in personal preference territory, I just happen to like the "method library" import syntax a little more than the extra exposure the folder groups brings.  In the end, I'm okay either way--that's just my input.

**Comment by jasonLaster** [comment]
Yep, don't want to slow you down w style bits.

**Comment by ahumphreys87** [comment]
I think this can be done non-breaking, should be push out to a 3.0.x?

**Comment by paulfalgout** [comment]
It isn't breaking, but I do think it might be highly related to the organizational structure of the documentation for `v3`.  Like the organization of the files _is_ essentially documentation.

**Comment by paulfalgout** [comment]
@marionettejs/marionette-core I recently moved a large project to ES6 import / export and we found a good way to reason about file structure, at least at first, was to make some simple rules such that "types" of files always export a certain way. so like everything we have in `/behaviors/` exports a `default` `behaviorClass`, but all `/util/` files export named exports even if there's only one thing in them.  

Essentially this was a strategy we used so that we could convert a bunch of files to ES6 import / export with multiple people without having to know what the dependencies exported, as someone else might have been working on that file.  But now that we're done, it actually seems quite useful going forward as well.

I think mixins for example should only `export default FooMixin;`
But perhaps `utils` and things like `bind-entity-events`should name their exports and should be somehow grouped together even if not in the same file.
